lib/galaxy/tools/toolbox/cache.py

```diff
@@ -7,9 +10,20 @@ class ToolCache(object):
     """
 
     def __init__(self):
+        self._hash_by_tool_paths = {}
         self._tools_by_path = {}
         self._tool_paths_by_id = {}
 
+    def cleanup(self):
+        """Remove uninstalled tools from tool cache if they are not on disk anymore or if their content has changed."""
+        clean_paths = {path: tool.all_ids for path, tool in self._tools_by_path.items() if not os.path.exists(path) or md5_hash_file(path) != self._hash_by_tool_paths[path]}
+        for path, tool_ids in clean_paths.items():
```

Comment:
I think `s/clean_paths/paths_to_cleanup/` in the previous 2 lines would be clearer.